sources/test/user/data.test.js

```diff
@@ -0,0 +1,195 @@
+const {beforeTests,afterTests, apolloServer, PERMISSION_DENIED_TEST} = require("../testHelper");
+
+// Pre and Post Test Scripts
+beforeAll(beforeTests);
+afterAll(afterTests);
+
+describe('Results: Users Queries and Mutations', () => {  
+
+  const { query, mutate } = apolloServer("a8mjiKYtt0PefnS524",["users.all","users.Auth","users.byName","users.byId","users.Update","users.Delete"]);
+  let testUser={
+    name:"Harish",
+    username:"HarishTeens"  
+  };
+  it('Get all Users', async () => {    
+      const GET_USERS = `
+        {
+          users{
+            ... on User{
+              name
+            }
+          }
+    
+        }
+      `;
+
+      const response = await query({ query: GET_USERS });
+      expect(response.data.users).toEqual([]);
+    });
+
+  it("Authenticate User(Sign up)",async ()=>{
+    const AUTH_USER = `
+      mutation {
+        authUser(user:{
+          username:"`+testUser.username+`",
+          gmailAuthMail:"arishh2@gmail.com",
+          name:"`+testUser.name+`"
+        }){
+          ... on User{
+            name
+            id,
+            username
+          }
+        }
+      }
+    `;
+
+    const response = await mutate({ mutation: AUTH_USER });
+    expect(response.errors).toEqual(undefined)
+    const userResponse= response.data.authUser;
+    testUser.id=userResponse.id;
+    expect(userResponse).toEqual(testUser);
+  })
+
+  it("Authenticate User(Sign in) + Level 1 check",async ()=>{
+    const testUser4={
```

Comment:
Do we need variable numbering like testuser4 and all? These are local variables local to each test block, so names can be resued I suppose

Reply:
Yes yes, that could be done.